pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -172,10 +174,13 @@ async def serve_requests(self, num_requests=None):
                 await self._on_serving_error(exc)
 
             finally:
-                if self._is_drain_needed:
+                if self._is_drain_needed and not self._discard_events:
                     result = self._call_drain_handler()
+                    self._discard_events = True
                     if asyncio.iscoroutine(result):
                         await result
+                self._is_drain_needed = False
```

Comment:
This is already set to False in `_call_drain_handler`,  no need to set it again

Reply:
@TomerShor we need to set it explicitly if wrapper is discarding events

Reply:
but why only drain if `not self._discard_events` in the first place?

I would move `self._discard_events = True` to `_on_drain_signal`, and it won't get new event anyway.
If we somehow get another drain signal - just drain again.

Reply:
@TomerShor I thought we discussed here https://github.com/nuclio/nuclio/pull/3120#discussion_r1452103978 that we don't want to drain again when wrapper is discarding events. 

Moved  `self._discard_events = True` to `_on_drain_signal`. Wrapper can get another drain signal from another partition which is mapped to the same worker.

Reply:
notice the you moved it to `_call_drain_handler` instead of `_on_drain_signal`. the latter is better IMO because we discard immediately when we get the signal.

And you are right, I got confused with what we decided. Let's do the check in `_on_drain_signal` as well, meaning:
When signal is received, if `discard_events == True` - ignore and do nothing. This way we keep the logic in the serve_requests loop simple (drain if needed and no other logic).